actor/builtin/miner/miner.go

```diff
@@ -809,11 +809,11 @@ func (ma *Actor) GetProvingPeriodStart(ctx exec.VMContext) (*types.BlockHeight,
 		return nil, errors.CodeError(err), err
 	}
 
-	return state.ProvingPeriodStart, 0, nil
+	return state.ProvingPeriodEnd, 0, nil
 }
 
 func currentProvingPeriodPoStChallengeSeed(ctx exec.VMContext, state State) (types.PoStChallengeSeed, error) {
-	bytes, err := ctx.SampleChainRandomness(state.ProvingPeriodStart)
+	bytes, err := ctx.SampleChainRandomness(state.ProvingPeriodEnd)
```

Comment:
Probably necessary to subtract the proving period duration here to retain semantics.